language/stdlib/modules/validator_config.move

```diff
@@ -76,44 +78,105 @@ module ValidatorConfig {
         move_to_sender<T>(
             T {
                 config: Config {
-                     consensus_pubkey,
-                     validator_network_signing_pubkey,
-                     validator_network_identity_pubkey,
-                     validator_network_address,
-                     fullnodes_network_identity_pubkey,
-                     fullnodes_network_address,
-                }
+                    consensus_pubkey: consensus_pubkey,
+                    network_signing_pubkey: *&validator_network_signing_pubkey,
+                    network_identity_pubkey: *&validator_network_identity_pubkey,
+                },
+                discovery_config: DiscoveryConfig {
+                    validator_network_identity_pubkey,
+                    validator_network_address,
+                    fullnodes_network_identity_pubkey,
+                    fullnodes_network_address,
+                },
+                delegation_enabled: false,
+                delegated_account: 0x0
             }
         );
     }
 
-    // Rotate a validator candidate's consensus public key. The change will not take effect until
-    // the next reconfiguration.
-    public fun rotate_consensus_pubkey(consensus_pubkey: vector<u8>) acquires T {
+    // If the sender of the transaction has a ValidatorConfig::T resource,
+    // this function will delegate management of the ValidatorConfig::T resource
+    // to a delegated_account.
+    public fun set_delegated_account(delegated_account: address) acquires T {
+        Transaction::assert(LibraAccount::exists(delegated_account), 5);
+        // check delegated address is different from transaction's sender
+        Transaction::assert(delegated_account != Transaction::sender(), 6);
+        let t_ref = borrow_global_mut<T>(Transaction::sender());
+        *(&mut t_ref.delegation_enabled) = true;
+        *(&mut t_ref.delegated_account) = delegated_account;
```

Comment:
I might be wrong, but I don't believe the `*(&mut ...)` is needed and this can instead be rewritten as: 
```suggestion
        t_ref.delegation_enabled = true;
        t_ref.delegated_account = delegated_account;
```